tools/swift-stdlib-tool/swift-stdlib-tool.cpp

```diff
@@ -670,583 +643,576 @@ -(NSString *)sst_command {
+    // one of the output buffers.
+
+    int errPipeReadFd = errPipe[1];
+    __block std::vector<uint8_t> stdErrData;
+    dispatch_semaphore_t gotStdErr = dispatch_semaphore_create(0);
+    dispatch_queue_t concurrentQueue = 
+        dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+    dispatch_async(concurrentQueue, ^{
+        stdErrData = readToEOF(errPipeReadFd);
+        dispatch_semaphore_signal(gotStdErr);
+    });
+
+    auto const stdOutData = readToEOF(outPipe[0]);
+
+    dispatch_semaphore_wait(gotStdErr, DISPATCH_TIME_FOREVER);
+    dispatch_release(gotStdErr);
+
+    int status = 0;
+    waitpid(childPid, &status, 0);
+    
+    // Task is finished and we have its stdout and stderr output.
+
+    // Print its stdout and stderr if it failed or we are verbose.
+    // (Print nothing by default because codesign is noisy.)
+
+    if (status  ||  Verbose > 1) {
+        fwrite(stdErrData.data(), stdErrData.size(), 1, stdout);
+        fwrite(stdOutData.data(), stdOutData.size(), 1, stdout);
     }
+    if (block) { block(stdOutData, stdErrData, status); }
+    return status;
 }
 
 
-void copyAndStripBitcode(NSString *src, NSString *dst)
+void copyAndStripBitcode(std::filesystem::path src, std::filesystem::path dst)
 {
     // -r removes bitcode
-    int err = xcrunToolCommand(@[@"bitcode_strip", src, @"-r", @"-o", dst]);
+    std::vector<std::string> commandAndArgs = {"bitcode_strip", src, "-r", "-o", dst};
+    int err = xcrunToolCommand(commandAndArgs);
 
     // Fail if bitcode_strip failed.
     if (err) {
         fail("Couldn't copy and strip bitcode %s to %s: bitcode_strip failed "
-             "with exit code %d", 
-             src.fileSystemRepresentation, dst.fileSystemRepresentation, err);
+             "with exit code %d", src.c_str(), dst.c_str(), err);
     }
 }
 
 
 void 
-copyFile(NSFileManager *fm, NSString *src, NSString *dst, bool stripBitcode)
+copyFile(std::filesystem::path src, std::filesystem::path dst, bool stripBitcode)
 {
     if (stripBitcode) {
         copyAndStripBitcode(src, dst);
     } else {
-        NSError *nserr;
-        if (![fm copyItemAtPath:src toPath:dst error:&nserr]) {
-            fail("Couldn't copy %s to %s: %s", 
-                 src.fileSystemRepresentation, 
-                 dst.fileSystemRepresentation, 
-                 nserr.localizedFailureReason.UTF8String);
+        if (rename(src.c_str(), dst.c_str()) != 0) {
+            fail("Couldn't copy %s to %s: %s", src.c_str(), dst.c_str(), strerror(errno));
         }
-    }
+   }
 }
 
+std::string uuidString(uuid_t uuid) {
+    char buffer[37];
+    uuid_unparse(uuid, buffer);
+    return buffer;
+}
 
-void copyLibraries(NSString *src_dir, NSString *dst_dir, 
-                   NSMutableDictionary *libs, bool stripBitcode)
+void copyLibraries(std::filesystem::path src_dir, std::filesystem::path dst_dir, 
+                   std::unordered_map<std::string,
+                   std::unordered_set<std::string>> libs,
```

Comment:
This is equivalent to `std::unordered_multimap`

Reply:
Does this routine mutate the map? It doesn't look like it, so take it by `const` ref please.

Reply:
I tried using a `std::unordered_multimap` but iterating through it in the same way and comparing to `dstUUIDs` below was a little messier.